pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -136,6 +136,12 @@ type ActualStateOfWorld interface {
 	// have no mountedPods. This list can be used to determine which volumes are
 	// no longer referenced and may be globally unmounted and detached.
 	GetUnmountedVolumes() []AttachedVolume
+
+	// GetActualPodVolumeSpecName generates and returns a map of pods in which map
+	// is indexed with pod uid and value is a map of the pod's mounted volume spec
+	// names. This map can be used to determine which pod and volume is currently
+	// in actual state of world based on pod uid and volume spec name.
+	GetActualPodVolumeSpecNames() map[string]map[string]bool
```

Comment:
Change it to only return a list pods' names. As long as the pod's name is in the list returned from actual or desired states, reconstruct function will skip reconstructing this pod. This logic works because volume lifecycle is the same as pod's. User's is acting on pods, not directly on volumes (at least currently we do not support updating volumes in a pod formally). If a pod is deleted, all its volumes information is deleted (reconstruct volume should proceed if the volume directory still exists on the disk). If the pod still exists in the states, all its volumes information should be available (no need for reconstruct process).